crates/ruff_workspace/src/options.rs

```diff
@@ -1327,6 +1329,7 @@ pub struct Flake8ImportConventionsOptions {
             scipy = "sp"
         "#
     )]
+    #[serde(deserialize_with = "deserialize_and_validate_aliases")]
     pub aliases: Option<FxHashMap<String, String>>,
```

Comment:
We would probably get better error messages if you create a newtype for `ModuleName` and `Alias` and implement `Deserialize` for them. I would expect that serde can then narrow the error message to the specific line rather than the entire table

Reply:
This PR does actually already seem to be a small improvement on the status quo when it comes to line numbers. Currently if I make this change to typeshed's pyproject.toml file:

```diff
diff --git a/pyproject.toml b/pyproject.toml
index 46014bcd2..927da6162 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -111,6 +111,9 @@ ignore = [
     "E501", # Line too long
 ]
 
+[tool.ruff.lint.flake8-import-conventions.aliases]
+"pandas" = 42
+
```

Then Ruff gives me this error message:

```
ruff failed
  Cause: Failed to parse /Users/alexw/dev/typeshed/pyproject.toml
  Cause: TOML parse error at line 27, column 1
   |
27 | [tool.ruff.lint]
   | ^^^^^^^^^^^^^^^^
invalid type: integer `42`, expected a string
```